### PR TITLE
criu: move workarea backup

### PR DESCRIPTION
**Pull request by sebratton**

### CODE REVIEW

**Comment by sebratton** [comment]
fixes #18425

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server [hunk 1]
```diff
@@ -1171,12 +1177,9 @@ restoreWorkarea()
   if [ ! -f "${SERVER_OUTPUT_DIR}/workarea/checkpoint/.restoreMarker" ]; then
     return 0
   fi
-  if [ ! -d "${SERVER_OUTPUT_DIR}/workarea-checkpoint" ]; then
+  if [ ! -d "${SERVER_OUTPUT_DIR}/workarea/checkpoint/workarea-checkpoint" ]; then
     return 1
   fi
-  if [ ! -d "${SERVER_OUTPUT_DIR}/workarea/checkpoint" ]; then
```
Comment (tjwatson) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:1177]: Did you think this was a duplicate check from above?  Or are you counting on the mv of checkpoint to fail below?  I guess having the mv fail will work also.

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server [hunk 2]
```diff
@@ -1185,10 +1188,9 @@ restoreWorkarea()
   fi
   rm -rf "${SERVER_OUTPUT_DIR}/workarea"
   # copy backup to workarea
-  cp -r --preserve=timestamps "${SERVER_OUTPUT_DIR}/workarea-checkpoint" "${SERVER_OUTPUT_DIR}/workarea"
+  cp -r --preserve=timestamps "${SERVER_OUTPUT_DIR}/checkpoint/workarea-checkpoint" "${SERVER_OUTPUT_DIR}/workarea"
   rc=$?
   # mv checkpoint directory back
-  mkdirs "${SERVER_OUTPUT_DIR}/workarea/"
```
Comment (tjwatson) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:1191]: I had this line here incase the `cp` above failed and the `workarea` dir didn't get created at all during the `cp` command.  I was trying hard to make sure the `checkpoint` directory got moved back under workarea even if the `cp` failed. so we didn't leave junk under the output directory.

**Review by tjwatson — COMMENTED** [review]
As we discussed, we could have the checkpoint workarea backup just be `workarea/checkpoint/workarea`. Adding the `-checkpoint` is a bit redundant.

**Comment by sebratton** [comment]
#build

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server (hunk 1 above)
Comment (sebratton) [reply dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:1177]: That check is now redundant. If the check in the **NEW code** immediately above passes:
```
  if [ ! -d "${SERVER_OUTPUT_DIR}/workarea/checkpoint/workarea-checkpoint" ]; then
    return 1
  fi
```
then `-d "${SERVER_OUTPUT_DIR}/workarea/checkpoint"` is true.

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server (hunk 2 above)
Comment (sebratton) [reply dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:1191]: Good point. I'll put back the line
```
mkdirs "${SERVER_OUTPUT_DIR}/workarea/"
```

**Comment by sebratton** [comment]
@tjwatson I updated the` checkpoint/workarea-checkpoint` name to be `checkpoint/workarea`.
As we discussed in slack, I'll may do a follow on change to reorganize probably following your suggestion:
` workarea/checkpoint/server/image|workarea`

**Comment by sebratton** [comment]
#build

**Review by tjwatson — APPROVED** [review]

**Pull request merged**